akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -934,6 +934,48 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
     new Flow(delegate.alsoToMat(that)(combinerToScala(matF)))
 
   /**
+   * Interleave represents deterministic merge of the given [[Source]] with elements of this [[Flow]].
+   * It takes `request` number of elements from this flow to emit downstream, then - same amount for given source,
+   * then repeat process from the beginning.
+   *
+   * Example:
+   * {{{
+   * Source<Integer, ?> src = Source.from(Arrays.asList(1, 2, 3))
+   * Flow<Integer, Integer, ?> flow = flow.interleave(Source.from(Arrays.asList(4, 5, 6, 7)), 2)
+   * src.via(flow) // 1, 2, 4, 5, 3, 6, 7
+   * }}}
+   *
+   * After [[Flow]] or [[Source]] is complete than all the rest elements will be emitted from the second one
+   *
+   * If this [[Flow]] or [[Source]] gets upstream error - stream completes with failure.
+   *
+   * '''Emits when''' element is available from current stream or from the given [[Source]] depending on what was pulled
+   *
+   * '''Backpressures when''' downstream backpressures
```

Comment:
Was recently thinking about that one in most docs; It does not sound very useful for fanout / fanin ops, 
since it does not explain if "any of the two backpressures" or if "both backpressure", can we reword this to explain the semantics we have here?